### DISCUSSION TITLE
shared users

### DISCUSSION THREAD

**Discussion by adamaen87** [arabcoders/watchstate#388] [General] [answered]
Hello,
thanks for this incredible tool.
I want to migrate from plex to emby, i have a lot of users who i share my plex with.
Is it possible to import shared users watchstate and export to emby?
when i configure my plex backend i only see local users, my main admin account and my children managed account.
Regards.

**Comment by arabcoders** [comment, marked as answer]
Hi, unfortunately friends accounts will not show up as they are separate accounts. You going to need their Plex token to access their sub account on your server. Sadly, this a limitation imposed by Plex.tv

**Reply by adamaen87** [reply]
Hi, thanks your answer, i already konw that and already got token access of my shared users, but how i can use it to transfer their watchstate from plex to their emby account (no problem with emby i can see all accounts :) ).
I don't understand how to do that,
Thanks for your help.

**Reply by arabcoders** [reply]
Yes you can. as watchstate is 1:1 app you just link different user and run the commands it's explained in the FAQ.md xD

**Reply by adamaen87** [reply]
Thanks, of course i already read the FAQ, search for discussions and issues. Maybe i am dumb but get trouble to understand how to do it.
First i need to create my plex backend with my main (admin) account token can only choose my main user?
Then add my emby backend (here i can select the user i want).
I already do that (plex as import and emby as export).
I don't understand the trick and how to do the link to select 1 of my shared user (with his token) and export it to emby.
If you can provide me some guidance it will be very kind of you.
Again thanks for your time.

**Reply by arabcoders** [reply]
You need different config and db per user. as the data cannot be imported on top of each other 

so if you just want a one time thing you just run container do one user stop it remove config dir. 

If you want to keep syncing, then you just have to run multiple containers that map 1:1 

the container itself doesn't really consume much memory except on import/export which is freed after the operation is done. 

I hope this clear things up a bit. If not hop on discord we can talk about strategy :)

**Reply by adamaen87** [reply]
The one time strategy is perfect for me as i want to get ride of plex. So run container, do one user, remove config and start again for another user is what i want. I just don't understand  how to select the correct user (with his token) in plex to export in emby, as i can't choose it when i create the plex backend.
Yes i can hop on discord :), i think i can get there in 1 hour ( a little more), on my lunch break.
Thanks.

**Comment by adamaen87** [comment]
With the help of the author on Discord this is the solution to migrate watchstate for a user you shared your server with:
- In order to do this you need the token of your Plex users, use this link, everything is well explained:
  https://www.plexopedia.com/plex-media-server/general/plex-token/  
- Have your watchstate docker running
- Then add your backends. Has i wanted to migrate form plex to emby, i added 2 backends. 1 for Plex and 1 for Emby.
   For Plex do it normally, when it will be the user selection you will only see your home users, select one, it doesn't matter, we will change it 
   later.
  Then i added the emby backend, and selected the right user i want to migrate.
- Once it's done, you must edit servers.yaml in /data/config of the watchstate container.
  you must have something like this:
plex:
  type: plex
  token: xxxxxxxxxx
  url: 'http://192.168.1.200:32400'
  uuid: xxxxxxxxxxxxxxxxxxxxxxx
  user: 4125556
  options:
    ADMIN_TOKEN: xxxxxxxxxxx
  import:
    enabled: true
    lastSync: 1703340958
  export:
    enabled: false
  webhook:
    match:
      user: true
      uuid: true
    token: xxxxxxxxxxxxxxxxxxxx

 Now you can delete the 2 lines: 
  options:
  ADMIN_TOKEN

 and then replace the second line "token:" with the token of the user you want to migrate watchstate.

You should then import data from plex: docker exec -ti watchstate console state:import -v --select-backends plex
Then export to emby: docker exec -ti watchstate console state:export -v --select-backends emby

If you a multiple users, just stop the container, delete files and restart.
